File: soap/Q2211_DataSumOfIntegers.java
import java.util.List;
import soap.q2211.generated.DataService;
import soap.q2211.generated.SoapDataService;

public class Q2211_DataSumOfIntegers {
    static final String STUDENT_CODE = "B21DCCN001";   // TODO
    static final String Q_CODE       = "irq9etkc";     // TODO

    public static void main(String[] args) {
        DataService service = new DataService();
        SoapDataService port = service.getSoapDataServicePort();

        // Phase 1 — getData
        List<Integer> data = port.getData(STUDENT_CODE, Q_CODE);
        System.out.println("data=" + data);

        int sum = data.stream().mapToInt(Integer::intValue).sum();
        System.out.println("sum=" + sum);

        // Phase 2 — submitDataInt (return void, server không trả status)
        port.submitDataInt(STUDENT_CODE, Q_CODE, sum);
        System.out.println("Đã submit sum=" + sum);
    }
}

File: soap/Q2211_DataSumOfIntegers.py
from zeep import Client

EXAM_IP      = "36.50.135.242"
STUDENT_CODE = "B22DCCN863"   # TODO
Q_CODE       = "irq9etkc"     # TODO
WSDL         = f"http://{EXAM_IP}:2221/DataService?wsdl"


def main():
    client = Client(WSDL)

    # Phase 1 — getData
    data = client.service.getData(STUDENT_CODE, Q_CODE)
    print("data=", data)

    total = sum(int(x) for x in data)
    print("sum=", total)

    # Phase 2 — submitDataInt (return void)
    client.service.submitDataInt(STUDENT_CODE, Q_CODE, total)
    print("Đã submit sum=", total)


if __name__ == "__main__":
    main()
